src/Pawn.cpp

```diff
@@ -0,0 +1,29 @@
+#include "..\include\Pawn.h"
+
+Pawn::Pawn(bool isWhite): Piece(isWhite)
+{
+}
+
+bool Pawn::isAcceptableMove(const Position& source, const Position& dest) const
+{
+    if (m_thisIsFirstMove)
+        if (this->getPieceColor())
+            return (dest.x == source.x + 2 || dest.x == source.x + 1) && dest.y == source.y;
+        else
+            return (dest.x == source.x - 2 || dest.x == source.x - 1) && dest.y == source.y;
+    else
+        if (this->getPieceColor())
+            return dest.x == source.x + 1 && dest.y == source.y;
+        else
+            return dest.x == source.x - 1 && dest.y == source.y;

```

Comment:
can be simplified and made more readable by breaking it down into smaller functions with clear responsibilities. This can improve code maintainability and readability.